docs/scripts.rst

```diff
@@ -0,0 +1,194 @@
+..
+  SPDX-FileCopyrightText: 2022 Nico Rikken <nico.rikken@fsfe.org>
+
+  SPDX-License-Identifier: CC-BY-SA-4.0
+
+==============
+Helper scripts
+==============
+
+This section contains scripts and snippets to help with the usage of reuse.
+Reuse already has a lot of functionality built-in, but there are still cases where reuse could use support from some external scripts.
+A typical example is adding SPDX headers based on the information in the version control system.
+This collection of scripts and snippets offers help for such situations.
+
+.. warning::
+  These scripts help you run reuse-tool against your codebase.
+  Automatically extracted information might not resemble the truth.
+  The correctness of these scripts is not guaranteed.
+  Use with caution and at your own risk.
+
+******************************
+Starting point of the codebase
+******************************
+
+The first code contribution can be a worthwile date to include in the copyright annotation.
+
+First commit
+============
+
+Git log can show summaries of all commits.
+At the moment of writing git does not allow selecting just one commit, so ``head`` is used to restrict the output.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --reverse --all | head -n 3
+  commit cdcea0887e0a85149f93e734b647301a16dd893e
+  Author: Carmen Bianca Bakker <carmenbianca@fsfe.org>
+  Date:   Tue Oct 10 18:27:11 2017 +0200
+
+.. SPDX-SnippetEnd
+
+Year of first commit
+====================
+
+With a custom format just the year of the first commit can be displayed.
+This output is convenient for use in larger scripts.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
```

Comment:
It's not defined, but wouldn't a REUSE compliant snippet also require a copyright line?

Reply:
I was thinking about it while writing, as you explicitly stated the copyright in your example in https://github.com/spdx/spdx-spec/pull/464 Yet if we have the same author for the scripts as the rest of the text, I don't see the need to duplicate the declaration. But that discussion might be better served for the SPDX repository. Also, the original author seems less relevant for a CC0 license.

I'm happy to change it if you still think it add value.

Reply:
True, I am also split. I'd say we can leave it for now, and see how the SPDX spec and our implementation evolves.